tests/queries/0_stateless/00652_mutations_alter_update.sh

```diff
@@ -123,5 +123,11 @@ ${CLICKHOUSE_CLIENT} --query="SELECT value2, materialized_value FROM alter_updat
 
 ${CLICKHOUSE_CLIENT} --query="ALTER TABLE alter_update DROP PARTITION 200001"
 
+${CLICKHOUSE_CLIENT} --query="SELECT '*** Test updating with SETTINGS update_allow_materialized_columns=1 ***'"
+${CLICKHOUSE_CLIENT} --query="INSERT INTO alter_update VALUES ('2000-01-01', 123, 'abc', 10), ('2000-01-01', 234, 'cde', 20)"
+${CLICKHOUSE_CLIENT} --query="ALTER TABLE alter_update \
+    UPDATE materialized_value = concat('updated_', toString(value2)) WHERE 1 \
+    SETTINGS update_allow_materialized_columns=1, mutations_sync=1"
```

Comment:
That's true, but those SETTINGS are volatile; once the mutation is persisted and loaded, it will start over from the global context. I'm not 100% happy about having to manually set `settings_ast` whenever a MutationCommands object is copied (easy to break in new code copying MutationCommands), so I'd be glad for any suggestions on how to improve that.